### PR TITLE
Upgrade to doctrine/lexer 3

### PR DESCRIPTION
**Pull request by greg0ire**
This should make it easier to install the project on projects with recent dependencies.

### CODE REVIEW

**Comment by greg0ire** [comment]
I would like to have a CI job on all Doctrine projects with the guides-cli running in it, so it's possible to forbid contributors to contribute invalid docs (based on the warnings output to `stderr`). This is one step towards doing that, another step will be to upgrade to Symfony 6

File: composer.lock [hunk 1]
```diff
@@ -1120,7 +1072,7 @@
         },
         {
             "name": "phpdocumentor/guides-cli",
-            "version": "dev-main",
+            "version": "dev-lexer-3",
```
Comment (greg0ire) [inline composer.lock:1075]: I don't know how to avoid this :thinking:

**Comment by greg0ire** [comment]
@wouterj I restored support for `doctrine/lexer` 2 so as not to cause issues similar to the one you are describing here.

File: packages/guides-restructured-text/composer.json [hunk 2]
```diff
@@ -22,7 +22,7 @@
     "minimum-stability": "stable",
     "require": {
         "php": "^8.1",
-        "doctrine/lexer": "^2.1",
+        "doctrine/lexer": "^3.0",
```
Comment (wouterj) [inline packages/guides-restructured-text/composer.json:25]:
```suggestion
        "doctrine/lexer": "^2.1 || ^3.0",
```

Btw, I think only supporting doctrine/lexer 3 is fine (as far as I can see in https://github.com/phpDocumentor/phpDocumentor , only the guides package is using Doctrine lexer).

**Review by wouterj — APPROVED** [review]

File: packages/guides-restructured-text/composer.json (hunk 2 above)
Comment (greg0ire) [reply packages/guides-restructured-text/composer.json:25]: You're right, I will change that.

Comment (greg0ire) [reply packages/guides-restructured-text/composer.json:25]: @wouterj I dropped `doctrine/lexer` 2 support again.

**Comment by jaapio** [comment]
Thanks for the upgrade!

**Review by jaapio — APPROVED** [review]

**Pull request merged**